email_notification/helpers.py

```diff
@@ -0,0 +1,44 @@
+from typing import List
+
+from django.db.models import Q
+
+from permissions.helpers import has_unit_permission
+from reservations.models import Reservation
+from spaces.models import Unit
+from users.models import ReservationNotification, User
+
+
+def get_staff_notification_recipients(
+    reservation: Reservation, notification_settings: List[ReservationNotification]
+) -> List[str]:
+    """
+    Get users with unit roles and notifications enabled, collect the ones that can manage relevant units,
+    have matching notification setting are not the reservation creator
+    """
+    notification_recipients: List[User] = []
+    reservation_units = reservation.reservation_unit.all()
+    units = Unit.objects.filter(reservationunit__in=reservation_units)
+
+    users = User.objects.filter(
+        Q(unit_roles__isnull=False) & ~Q(reservation_notification="NONE")
```

Comment:
service_sector_roles perhaps needs also? Or is that something that is "too high" in hierarchy and don't want mails?

Reply:
It was specified so that notifications are sent only to people who have unit-level roles. From the ticket:
> jos oikeus on has_service_sector_permission kautta eli käyttäjällä on oikeudet toimipisteeseen sen toimialan kautta yleisesti, niin tällöin ilmoituksia ei siitä myöskään lähetetä (ellei lisäksi ole unit-tason permissio).